src/sentry/spans/consumers/process/factory.py

```diff
@@ -47,11 +47,13 @@ def _build_snuba_span(relay_span: Mapping[str, Any]) -> MutableMapping[str, Any]
         or DEFAULT_SPAN_RETENTION_DAYS
     )
 
+    span_data: Mapping[str, Any] = relay_span.get("data", {})
+
     snuba_span: MutableMapping[str, Any] = {}
     snuba_span["description"] = relay_span.get("description")
     snuba_span["event_id"] = relay_span["event_id"]
     snuba_span["exclusive_time_ms"] = int(relay_span.get("exclusive_time", 0))
-    snuba_span["group_raw"] = "0"
+    snuba_span["group_raw"] = span_data.get("span.group", "0")
```

Comment:
I looked at how Snuba consumer is currently doing it. The code is [here](https://github.com/getsentry/snuba/blob/master/snuba/datasets/processors/spans_processor.py#L301). Now I remember that we store the hash of both scrubbed description and unscrubbed description. `group_raw` stores the hash of the unscrubbed description.
```suggestion
    snuba_span["group_raw"] = relay_span.get("hash", "0")
```

Reply:
There's no field `hash` in the span I'm getting from relay (https://github.com/getsentry/relay/blob/master/relay-event-schema/src/protocol/span.rs#L14).

We're extracting the `span.group` tag https://github.com/getsentry/relay/blob/8048993b105d470d2f55af1ae5fe027ae16c19a7/relay-event-normalization/src/normalize/span/tag_extraction.rs#L287 and all these tags are set in the `data` field https://github.com/getsentry/relay/blob/8048993b105d470d2f55af1ae5fe027ae16c19a7/relay-event-normalization/src/normalize/span/tag_extraction.rs#L172

Reply:
I see. Do we know who adds the field `hash` in the transaction event payload today? Is it the SDK, Relay, ingest-transaction consumer? If it is the ingest-transaction consumer we would probably need to copy over the same mechanism as used by the consumer. If it is before ingest-transaction consumer then it would need to be encoded in the protocol.